src/main/java/me/xmrvizzy/skyblocker/skyblock/EtherwarpOverlay.java

```diff
@@ -0,0 +1,56 @@
+package me.xmrvizzy.skyblocker.skyblock;
+
+import org.lwjgl.glfw.GLFW;
+
+import com.mojang.blaze3d.systems.RenderSystem;
+
+import me.xmrvizzy.skyblocker.config.SkyblockerConfig;
+import me.xmrvizzy.skyblocker.skyblock.item.PriceInfoTooltip;
+import me.xmrvizzy.skyblocker.utils.RenderHelper;
+import me.xmrvizzy.skyblocker.utils.Utils;
+import net.fabricmc.fabric.api.client.rendering.v1.WorldRenderContext;
+import net.fabricmc.fabric.api.client.rendering.v1.WorldRenderEvents;
+import net.minecraft.block.BlockState;
+import net.minecraft.block.Blocks;
+import net.minecraft.client.MinecraftClient;
+import net.minecraft.client.util.InputUtil;
+import net.minecraft.item.ItemStack;
+import net.minecraft.nbt.NbtCompound;
+import net.minecraft.util.hit.BlockHitResult;
+import net.minecraft.util.hit.HitResult;
+import net.minecraft.util.math.BlockPos;
+
+public class EtherwarpOverlay {
+	private static final MinecraftClient CLIENT = MinecraftClient.getInstance();
+	private static final float[] COLOR_COMPONENTS = { 118f / 255f, 21f / 255f, 148f / 255f };
+	
+	public static void init() {
+		WorldRenderEvents.AFTER_TRANSLUCENT.register(EtherwarpOverlay::renderEtherwarpOverlay);
+	}
+
+	private static void renderEtherwarpOverlay(WorldRenderContext wrc) {
+		if (Utils.isOnSkyblock() && SkyblockerConfig.get().general.etherwarpOverlay) {
+			ItemStack heldItem = CLIENT.player.getMainHandStack();
+			String itemId = PriceInfoTooltip.getInternalNameFromNBT(heldItem);
+			NbtCompound nbt = heldItem.getNbt();
+			
+			if (itemId != null && (itemId.equals("ASPECT_OF_THE_VOID") || itemId.equals("ASPECT_OF_THE_END")) && (nbt != null && nbt.getCompound("ExtraAttributes").getInt("ethermerge") == 1) && InputUtil.isKeyPressed(CLIENT.getWindow().getHandle(), GLFW.GLFW_KEY_LEFT_SHIFT)) {
```

Comment:
instead of left shift shouldn't we check on the sneak key ?

Reply:
```suggestion
			if (itemId != null && (itemId.equals("ASPECT_OF_THE_VOID") || itemId.equals("ASPECT_OF_THE_END")) && (nbt != null && nbt.getCompound("ExtraAttributes").getInt("ethermerge") == 1) && CLIENT.options.sneakKey.isPressed()) {
```
this is a way to do it, we could also check if the player is sneaking